src/raster/processing/dem/aspect.rs

```diff
@@ -0,0 +1,125 @@
+use super::options::common_dem_options;
+use crate::cpl::CslStringList;
+use crate::raster::processing::dem::DemSlopeAlg;
+use std::num::NonZeroUsize;
+
+/// Configuration options for [`aspect()`][super::aspect()].
+#[derive(Debug, Clone, Default)]
+pub struct AspectOptions {
+    input_band: Option<NonZeroUsize>,
+    compute_edges: bool,
+    output_format: Option<String>,
+    additional_options: CslStringList,
+    algorithm: Option<DemSlopeAlg>,
+    zero_for_flat: Option<bool>,
```

Comment:
I'm inclined to say we should avoid `Option` for most, if not all of these. I understand that the default might change, but if `-zero_for_flat` ever becomes the default, there will be a new `-non_zero_for_flat` and we'll have to update the code anyway. That is,

```rust
        if self.zero_for_flat == Some(true) {
            opts.add_string("-zero_for_flat").unwrap();
        }
```

already assumes that the default is `false`.

But we can probably change them later, you must be sick of this PR by now.